aesthetic_ants/penguin/tile_layer.py

```diff
@@ -0,0 +1,113 @@
+import pyglet
+
+from math import floor
+from pyglet.graphics import Batch
+
+from .constants import CollisionType, TileType
+from .object import Object
+from .resources import WATER_TILE, ICE_TILE, WEAK_ICE_TILE, WALL_TILE
+
+TILE_SIZE = 16  # tile size in pixels
+
+
+TILE_SERIALIZATION_MAP = {
+    "W": TileType.WALL,
+    "I": TileType.ICE,
+    "E": TileType.WEAK_ICE,
+    "A": TileType.WATER
+}
+
+
+class Tile(pyglet.sprite.Sprite):
+    """
+    Represents an individual tile in a tiled layer
+    """
+    def __init__(self, tile, *args, **kwargs):
+        self.tile_type = tile
+        super().__init__(*args, **kwargs)
+
+
+class TileLayer(Object):
+    """
+    A class that manages a large array of tile sprites
+    """
+    collision_type = CollisionType.TILE_LAYER
+
+    def __init__(self, x, y):
+        self.tile_images = {
+            TileType.WATER: WATER_TILE,
+            TileType.ICE: ICE_TILE,
+            TileType.WEAK_ICE: WEAK_ICE_TILE,
+            TileType.WALL: WALL_TILE
+        }
+
+        self.x = x
+        self.y = y
+
+        self.batch = Batch()
+
+        # Tiles is in the format [y][x]
+        self.tiles = []
+
+    def set_tile(self, tile_x: int, tile_y: int, tile_type: TileType):
+        """
+        Creates a new tile at a given position in the tile grid
+        """
+        if self.tiles[tile_y][tile_x]:
+            self.tiles[tile_y][tile_x].batch = None
+            self.tiles[tile_y][tile_x].delete()
+
+        self.tiles[tile_y][tile_x] = Tile(tile_type,
+                                          self.tile_images[tile_type],
+                                          x=tile_x * TILE_SIZE + self.x,
+                                          y=tile_y * TILE_SIZE + self.y,
+                                          batch=self.batch)
+
+    def update(self, dt, **kwargs):
+        pass
+
+    def draw(self):
+        self.batch.draw()
+
+    def load_tiles(self, tile_filename):
+        """Loads a tile map from a level file. Levels are 40x30 tiles by default"""
+        with open(tile_filename, 'r') as level_file:
```

Comment:
It might be a good idea to delete the old tiles (probably by making a delete method or something) before loading this in case this gets called multiple times.